core/kernel/tee_ta_manager.c

```diff
@@ -957,20 +957,22 @@ static void tee_ta_update_session_utime(bool suspend)
 	now = read_cntpct();
 
 #if defined(CFG_TA_GPROF_SUPPORT)
-	gprof_update_session_utime(suspend, s, now);
+	if (flag & GPROF_UPDATE_FLAG)
+		gprof_update_session_utime(suspend, s, now);
 #endif
 #if defined(CFG_TA_FTRACE_SUPPORT)
-	ftrace_update_session_utime(suspend, s, now);
+	if (flag & FTRACE_UPDATE_FLAG)
+		ftrace_update_session_utime(suspend, s, now);
```

Comment:
I suggest `ftrace_update_times()`, and move this out so it does not have anything to do with any function called `*session_utime*()`.

Reply:
Okay.